pvlib/iam.py

```diff
@@ -274,6 +274,114 @@ def martin_ruiz(aoi, a_r=0.16):
+        This is an empirical dimensionless parameter. Values of a_r are
+        generally on the order of 0.08 to 0.25 for flat-plate PV modules.
+        a_r must be a positive numeric scalar.
+
+    c1, c2 : numeric scalar
+        Two fitting parameters for the expressions that approximate the
+        integral of diffuse irradiance coming from different directions.
+        c1 is constant at 4 / 3 / pi (0.4244) and c2 varies with a_r.  The
+        calculation of c2 from a_r is according to [3].
+
+    Returns
+    -------
+    iam_sky : numeric
+        The incident angle modifier for sky diffuse
+
+    iam_ground : numeric
+        The incident angle modifier for ground-reflected diffuse
+
+    Notes
+    -----
+    Sky and ground modifiers are complementary: iam_sky for tilt = 30 is
+    equal to iam_ground for tilt = 180 - 30.  For vertical surfaces,
+    tilt = 90, the two factors are equal.
+
+    References
+    ----------
+    [1] N. Martin and J. M. Ruiz, "Calculation of the PV modules angular
+    losses under field conditions by means of an analytical model", Solar
+    Energy Materials & Solar Cells, vol. 70, pp. 25-38, 2001.
+
+    [2] N. Martin and J. M. Ruiz, "Corrigendum to 'Calculation of the PV
+    modules angular losses under field conditions by means of an
+    analytical model'", Solar Energy Materials & Solar Cells, vol. 110,
+    pp. 154, 2013.
+
+    [3] "IEC 61853-3 Photovoltaic (PV) module performance testing and energy
+    rating - Part 3: Energy rating of PV modules". IEC, Geneva, 2018.
+
+    See Also
+    --------
+    iam.martin_ruiz
+    iam.physical
+    iam.ashrae
+    iam.interp
+    iam.sapm
+    '''
+    # Contributed by Anton Driesse (@adriesse), PV Performance Labs. Oct. 2019
+
+    if isinstance(surface_tilt, pd.Series):
+        out_index = surface_tilt.index
+    else:
+        out_index = None
+
+    surface_tilt = np.asanyarray(surface_tilt)
+
+    with np.errstate(invalid='ignore'):
+        # undo possible surface rotations
+        surface_tilt = (surface_tilt + 180) % 360 - 180
+
+        # flip backward tilts forward
+        surface_tilt = np.abs(surface_tilt)
+
+        # avoid undefined results for horizontal or upside-down surfaces
+        small_angle = 1e-06
+        surface_tilt = np.clip(surface_tilt, small_angle, 180 - small_angle)
+
+    if np.any(np.less_equal(a_r, 0)):
+        raise ValueError("The parameter 'a_r' cannot be zero or negative.")
+
+    if c2 is None:
+        # This equation is from [3] Sect. 7.2
+        c2 = 0.5 * a_r - 0.154
+
+    from numpy import pi, radians, sin, cos, exp
+
+    beta = radians(surface_tilt)
+
+    with np.errstate(invalid='ignore'):
+        # because sin(pi) isn't exactly zero
+        sin_beta = np.where(surface_tilt < 90, sin(beta), sin(pi - beta))
+
+        trig_term_sky = sin_beta + (pi - beta - sin_beta) / (1 + cos(beta))
+        trig_term_gnd = sin_beta +      (beta - sin_beta) / (1 - cos(beta))
```

Comment:
```suggestion
        trig_term_gnd = sin_beta +      (beta - sin_beta) / (1 - cos(beta))  # noqa: E222
```

Reply:
Let's see if it complains about the line length now...

Reply:
Ok, I can silence that one too.